### PR TITLE
[DELIVERABLE 3] Hazard Analysis: Critical Assumptions 

### LINKED ISSUE #159: [HA] Critical Assumptions

### PR DESCRIPTION
**Pull request by tbrar06**
## Description
Added Critical Assumptions for Hazard Analysis 
## How Has This Been Tested?
Compiled locally 

Closes #159

### CODE REVIEW

File: docs/HazardAnalysis/HazardAnalysis.tex [hunk 1]
```diff
@@ -158,10 +158,16 @@ \subsubsection*{Web Client}
 
 \section{Critical Assumptions}
 
-\wss{These assumptions that are made about the software or system.  You should
-minimize the number of assumptions that remove potential hazards.  For instance,
-you could assume a part will never fail, but it is generally better to include
-this potential failure mode.}
+\begin{itemize}
+    \item The Energy Measurement Model will provide accurate and consistent energy consumption metrics across different platforms (Windows, macOS, Linux). There are no discrepancies in measurements due to platform differences that could result in ineffective refactoring.
+    \item Refactoring will not introduce new bugs or alter the intended functionality of the original Python code.
```
Comment (Sevhena) [inline docs/HazardAnalysis/HazardAnalysis.tex:163]: I don't know if this assumption is right to make. The comment the prof left on the doc mentioned that we shouldn't make assumptions that negate risks. Not to mention we have a testing module meaning we clearly assume that there _could_ be bugs

File: docs/HazardAnalysis/HazardAnalysis.tex [hunk 2]
```diff
@@ -158,10 +158,16 @@ \subsubsection*{Web Client}
...
+    \item The Testing Module is provided with automated tests that have enough coverage to detect post refactoring bugs, functionality regressions, etc.
+    \item Code smells identified by the Refactoring Module always involve code that could be more energy efficient.
+    \item Custom-made refactoring strategies and Rope are capable of generating effective and correct refactoring.
+    \item Sufficient data sets are available for the reinforcement learning model to provide increasingly accurate and efficient refactoring suggestions over time.
+    \item GitHub Actions, which is a third-party dependency for the DevOps integration, is not suspended for a prolonged period of time.
+    \item The tool can handle large codebases without requiring excessive computational resources.
```
Comment (Sevhena) [inline docs/HazardAnalysis/HazardAnalysis.tex:169]: I'm also a bit iffy about this one. It is one of our non-functionals and there is a very real chance that this assumption will be proved false more often than not.

File: docs/HazardAnalysis/HazardAnalysis.tex (hunk 1 above)
Comment (tbrar06) [reply docs/HazardAnalysis/HazardAnalysis.tex:163]: I agree, going to remove this assumption

File: docs/HazardAnalysis/HazardAnalysis.tex (hunk 2 above)
Comment (tbrar06) [reply docs/HazardAnalysis/HazardAnalysis.tex:169]: Thanks, this will also be removed

**Review by mmyaaaaa — APPROVED** [review]

**Review by Ayushi1972 — APPROVED** [review]
Looks good

**Pull request merged**